File: lokask-backend/backend/internal/domain/booking.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type BookingEntry struct {
	ID           uuid.UUID `db:"id" json:"id"`
	ConsultantID uuid.UUID `db:"consultant_id" json:"consultant_id"`
	UserID       uuid.UUID `db:"user_id" json:"user_id"`

	// Time slots
	StartTime time.Time `db:"start_time" json:"start_time"`
	EndTime   time.Time `db:"end_time" json:"end_time"`

	// Management
	ServiceType string  `db:"service_type" json:"service_type"`
	Status      string  `db:"status" json:"status"` // pending, confirmed, cancelled
	TotalPrice  float64 `db:"total_price" json:"total_price"`
	UserNotes   string  `db:"user_notes" json:"user_notes"`

	CreatedAt time.Time `db:"created_at" json:"created_at"`
	UpdatedAt time.Time `db:"updated_at" json:"updated_at"`
}

type CreateBookingRequest struct {
	ConsultantID string  `json:"consultant_id"`
	StartTime    string  `json:"start_time"`   // Received as ISO string from frontend
	ServiceType  string  `json:"service_type"` // e.g., 'video_call'
	UserNotes    string  `json:"user_notes"`
	TotalPrice   float64 `json:"total_price"`
}

File: lokask-frontend/src/types/booking.ts
export type ServiceType = "chat_only" | "video_call" | "voice_call" | "itinerary_review";
export type BookingStatus = "pending" | "confirmed" | "completed" | "cancelled";

export interface Booking {
  id: string;
  consultant_id: string;
  user_id: string;
  
  // Time slots (coming from PostgreSQL as ISO strings)
  start_time: string;
  end_time: string;
  
  // Management
  service_type: ServiceType;
  status: BookingStatus;
  total_price: number;
  user_notes: string;
  
  created_at: string;
  updated_at: string;

  // View fields joined from the database
  traveller_name?: string;
  traveller_avatar?: string;
  traveller_location?: string;
  consultant_name?: string;
  consultant_avatar?: string;
  consultant_city?: string;
}

export interface CreateBookingRequest {
  consultant_id: string;
  start_time: string; // Must be sent as an ISO string
  service_type: string;
  user_notes: string;
  total_price: number;
}